html_app/components/FACSModelFactory.js

```diff
@@ -9,58 +9,94 @@ scb.components.FACSModelFactory = function scb_components_FACSModelFactory(model
 
   if (scb.utils.isDefined(model.dna)) {
     self.dna = function(state) {
-      var t = template;
       var m = model.dna;
-      if (m.parser_simple) {
-        var facs_lane = state.facs_lane;
-        var cell_treatment = facs_lane.cell_treatment;
-        var drug_treatments = cell_treatment.treatment_list.list;
-        var duration = drug_treatments[0].duration;
-        var shape = '';
-        var facs_state = {
-          cell_line: function(str) {
-            return str == cell_treatment.cell_line
-          },
-          temperature: function(str) {
-            return str == drug_treatments[0].temperature
-          },
-          condition: function(str) {
-            return str == facs_lane.conditions;
-          },
-          duration: function(str) {
-            return str == duration;
-          },
-          drug_id: function(str) {
-            var any = false;
-            _.each(drug_treatments, function(dt) {
-              _.each(dt.drug_list.list, function(drug) {
-                any |= (drug.drug_id == str);
+      if (template.model.facs.is_ab) {
+        if (m.parser_simple) {
+          var facs_lane = state.facs_lane;
+          var cell_treatment = facs_lane.cell_treatment;
+          var facs_state = {
+            identifier: function (str) {
+              return str == cell_treatment.identifier;
+            },
+            treatment: function (str) {
+              return str == facs_lane.live;
+            },
+            analysis: function (str) {
+              return str == facs_lane.kind;
+            },
+            condition: function (str) {
+              return str == facs_lane.conditions;
+            }
+          };
+          /* Iterating over the rules */
+          var matching_rule = _.find(m.parser_simple, function (rule) {
+
+              var matches = true;
+              _.each(_.keys(facs_state), function (property) {
```

Comment:
FYI you can use `_.each` to iterate over the keys and values of an object, like this: `_.each(obj, function(value, key) { ... })`